packages/amplify-graphql-model-transformer/src/resources/rds-model-resource-generator.ts

```diff
@@ -130,3 +140,25 @@ export class RdsModelResourceGenerator extends ModelResourceGenerator {
     });
   }
 }
+
+/**
+ * Resolves the layer version using an appropriate strategy for the current context. In the Gen1 CLI flow, the transform-graphql-schema-v2
+ * buildAPIProject function retrieves the latest layer version from the S3 bucket. In the CDK construct, such async behavior at synth time
+ * is forbidden, so we use an AwsCustomResource to resolve the latest layer version. The AwsCustomResource does not work with the CLI custom
+ * synth functionality, so we fork the behavior at this point.
+ *
+ * Note that in either case, the returned value is not actually the literal layer ARN, but rather a reference to be resolved at deploy time:
+ * in the CLI case, it's the resolution of the SQLLayerMapping; in the CDK case, it's the 'Body' response field from the AwsCustomResource's
+ * invocation of s3::GetObject.
+ */
+const resolveLayerVersion = (scope: Construct, context: TransformerContextProvider): string => {
+  let layerVersionArn: string;
+  if (context.rdsLayerMapping) {
+    setRDSLayerMappings(scope, context.rdsLayerMapping);
+    layerVersionArn = Fn.findInMap(ResourceConstants.RESOURCES.SQLLayerMappingID, Fn.ref('AWS::Region'), 'layerRegion');
+  } else {
+    const layerVersionCustomResource = createLayerVersionCustomResource(scope);
+    layerVersionArn = layerVersionCustomResource.getResponseField('Body');
+  }
+  return layerVersionArn;
```

Comment:
super np, but I'd prefer to avoid the `let` entirely, and just return the right type based on 
```suggestion
  if (context.rdsLayerMapping) {
    setRDSLayerMappings(scope, context.rdsLayerMapping);
    return Fn.findInMap(ResourceConstants.RESOURCES.SQLLayerMappingID, Fn.ref('AWS::Region'), 'layerRegion');
  }
  const layerVersionCustomResource = createLayerVersionCustomResource(scope);
  return layerVersionCustomResource.getResponseField('Body');
};
```

Reply:
Ah, I follow you. I'll add that to a future change